src/v/storage/ntp_config.h

```diff
@@ -309,6 +311,14 @@ class ntp_config {
                           : default_iceberg_enabled;
     }
 
+    bool cloud_topic_enabled() const {
```

Comment:
Similar here. Should be hard coded to false for internal topics for now, right? https://github.com/redpanda-data/redpanda/blob/d17d5b687e88fd37fdf9e24dc3253cb98038d22d/src/v/storage/ntp_config.h#L265

Reply:
i don't know. can you say more? this pr is intended to disallow usage of cloud topics except for in development, and nothing more. do we need more?

Reply:
it depends very much on the answer to the question from https://github.com/redpanda-data/redpanda/pull/23716#discussion_r1862259806